keps/sig-network/2433-topology-aware-hints/README.md

```diff
@@ -369,6 +371,43 @@ of the following scenarios:
 2. A new Node results in a Service that is able to achieve an endpoint
    distribution below 20% for the first time.
 
+### Manual enforcement of Topology Aware Routing by zone
+
+The algorithm proposed for mitigating the risk of overload works best with large
+numbers of endpoints and nodes, such that the percentage thresholds are unlikely
+to be broached with service restarts and node additions/removals within zones.
+There are a number of use cases however where the proposed algorithm for automatic
+assignments of zone hints is not suitable:
+
+1. Small multi-zone clusters with a low number of zones and endpoints. For 
+example a rolling restart with a service in a 3 zone cluster with 2-2-2 
+endpoint distribution will cause the proposed algorithm to remove hints for the 
+service during the restart, this could have severe impact for high latency 
```

Comment:
this is an interesting problem @robscott ,  the time "without hints" here will be the sum of all the restarts in a deployment rolling update , right?

however, this should be really a problem for long lived connection initiated during this window, no?

Reply:
Not really, this is a real issue for multi-region clusters with non-trivial latency between regions. For example we have a cluster across NYC/LON/India. In normal operation all DNS requests are sent to local DNS instances of CoreDNS. If we restart the CoreDNS service though then for the entire rolling restart process DNS requests are sent randomly to different CoreDNS instances globally, this could make DNS requests go from 20ms to 500ms depending on the path chosen.